### PR TITLE
samples: net: lwm2m_client: Split sensor modules out from main

### LINKED ISSUE #19356: LwM2M sample reorganization: split out LwM2M source into object-based .c files
**Is your enhancement proposal related to a problem? Please describe.**
The LwM2M sample has become fairly complex over the last 2 years.  I propose to break out each LwM2M object / configuration into it's own source file.

**Describe the solution you'd like**
Each object-based .c file could be located under the `src/lwm2m/` directory.  This way new users of the code can easily locate things they are interested in like the `lwm2m_button.c` file or `lwm2m_firmware.c`.

**Describe alternatives you've considered**
The sample is only going to get more complex as I add support for different network type, so this is needed prior to those changes.

### PR DESCRIPTION
**Pull request by SeppoTakalo**
Each module deserves their own file so we don't clutter the main application.

Fixes #19356

### CODE REVIEW

File: samples/net/lwm2m_client/src/modules.h [hunk 1]
```diff
@@ -0,0 +1,9 @@
+#ifndef _MODULES_H
```
Comment (rlubos) [inline samples/net/lwm2m_client/src/modules.h:1]: License header is missing.

**Review by rlubos — COMMENTED** [review]
Looks good, but please address the compliance issues.

File: samples/net/lwm2m_client/src/modules.h (hunk 1 above)
Comment (SeppoTakalo) [reply samples/net/lwm2m_client/src/modules.h:7]: Fixed.

**Review by rlubos — APPROVED** [review]

**Review by jukkar — APPROVED** [review]
LGTM

**Pull request merged**